numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -38,6 +38,39 @@
 #include "npy_cblas.h"
 #include "npy_buffer.h"
 
+
+/*
+ * Define a stack allocated dummy array with only the minimum information set:
+ *   1. The descr, the main field interesting here.
+ *   2. The flags, which are needed for alignment; The OWNDATA flag is
+ *      cleared (this is correct and required for 3.).
+ *   3. base, reference count, and type are set because these are used
+ *      during setting of the base object. Setting all ensures that
+ *      the dummy array will be skipped when setting the base.
+ *      (i.e. the dummy arrays `base` is used instead).
+ *
+ * In debug mode fill in the `data` field to ensure a crash if the object
+ * is incorrectly deallocated.
+ */
+#define _MAKE_DUMMY_STACK_ARRAY(arr, dummy_array)                   \
+        PyArrayObject_fields __dummy_array_fields;                  \
```

Comment:
Good practice to wrap this in `do { ... } while(0)` so it consumes semicolons correctly.

Reply:
Oh, that won't work, because this is on the stack - nevermind

Reply:
Hmmm, wouldn't that kill scoping of the PyArrayObject_fields structure? To be honest, probably depends on what the C standard says, so no clue.

Reply:
You're correct